### ISSUE TITLE
Block pattern inserter modal: adding a pattern moves focus outside of the Modal

### ISSUE THREAD

**Issue by ciampo** [WordPress/gutenberg#64853]
### What

When clicking on a pattern in the block patterns inserter modal, focus is moved outside of the modal itself.

https://github.com/user-attachments/assets/5bcacc13-7562-43e9-8d5e-dc7647e9255e

Not sure if this has always been a problem, or if it's a regression caused by other changes.

### Proposed solution

Losing focus while the modal is open is considered an anti-pattern.

There are two alternative solutions:

- Keep the focus on the selected pattern in the modal
- Close the modal when moving focus outside of it

**Comment by ciampo** [comment]
cc @ntsekouras and @scruffian who are among the last folks to have worked on this

**Comment by ntsekouras** [comment]
If I remember correctly when it was first implemented the inserter used to close and there was no focus loss issue. This seems related to the changes to keep the inserter open after an insertion. Maybe @MaggieCabrera can ping some folks to see if that's the case?

**Comment by ntsekouras** [comment]
@ciampo can you retest this one. I couldn't reproduce it right now.

**Comment by ciampo** [comment]
Can't repro either. We can re-open in case we can repro again
